src/PlatformSockets.c

```diff
@@ -302,7 +306,7 @@ int setSocketNonBlocking(SOCKET s, bool enabled) {
 #elif defined(O_NONBLOCK)
     return fcntl(s, F_SETFL, (enabled ? O_NONBLOCK : 0) | (fcntl(s, F_GETFL) & ~O_NONBLOCK));
 #elif defined(FIONBIO)
-    int val = enabled ? 1 : 0;
+    unsigned long val = enabled ? 1 : 0;
     return ioctlsocket(s, FIONBIO, &val);
```

Comment:
I don't think this part is correct.

It will work on Windows, because Windows uses the LLP64 data model (`sizeof(long) == sizeof(int)`). It won't be correct on Linux which uses LP64 (`sizeof(long) > sizeof(int)`).

What about something like:
```
#ifdef LC_WINDOWS
    u_long val = enabled ? 1 : 0;
#else
    int val = enabled ? 1 : 0;
#endif
    return ioctlsocket(s, FIONBIO, &val);
```